examples/recipes/output_csv_stream_asgi.py

```diff
@@ -0,0 +1,33 @@
+import csv
+
+
+class Report:
+    class PseudoTextStream:
+        def __init__(self):
+            self.clear()
+
+        def clear(self):
+            self.result = []
+
+        def write(self, data):
+            self.result.append(data.encode())
+
+    async def fibonacci_generator(self, n=1000):
+        stream = self.PseudoTextStream()
+        writer = csv.writer(stream, quoting=csv.QUOTE_NONNUMERIC)
+        writer.writerow(('n', 'Fibonacci Fn'))
+
+        previous = 1
+        current = 0
+        for i in range(n + 1):
+            writer.writerow((i, current))
+            previous, current = current, current + previous
+
+            for chunk in stream.result:
+                yield chunk
```

Comment:
yield from? this seems the only difference from the wsgi one (other than the async bits)

Reply:
See [the note on the recipe itself](https://falcon.readthedocs.io/en/latest/user/recipes/output-csv.html#streaming-large-csv-files-on-the-fly); also here: [Asynchronous `yield from`](https://peps.python.org/pep-0525/#asynchronous-yield-from).

Reply:
interesting. that's not actually async though, but I guess it doesn't matter to the parser.